src/core/backbone_router/local.cpp

```diff
@@ -194,7 +220,30 @@ void Local::SetState(BackboneRouterState aState)
 {
     VerifyOrExit(mState != aState, OT_NOOP);
 
+    if (mState == OT_BACKBONE_ROUTER_STATE_DISABLED)
+    {
+        // Subscribe All Network Backbone Routers Multicast Address for both Secondary and Primary state.
+        mAllNetworkBackboneRouters.GetAddress().SetMulticastNetworkPrefix(Get<Mle::MleRouter>().GetMeshLocalPrefix());
```

Comment:
Note that previously when addresses were in `ot::Mle::Mle`, there are `SetMeshLocalPrefix()` which is mainly used to remove/unsubscribe addresses and update the prefix,  `ApplyMeshLocalPrefix()` which is mainly used to add/subscribe addresses.

For the other components, like `BackboneRouter::Local`, or the `MeshCop::Commissioner` (see #4894), we may follow similar ways to introduce `SetMeshLocalPrefix()` and `ApplyMeshLocalPrefix()`. However I think it might not be necessary - for example, what's the point of updating `mAllNetworkBackboneRouters`  if `BACKBONE_ROUTER` is not enabled,  or updating `mBackboneRouterPrimaryAloc` if it is not primary?

So in this PR, or in #4894, only `ApplyMeshLocalPrefix()`  is introduced, and only update the prefix when necessary, whenever relative address need to added/subscribed, always construct the address from existing mesh local prefix.

Reply:
Currently, `mAllNetworkBackboneRouters` are returned in a public method of `BackboneRouter::Local`
```
    /**
     * This method returns a reference to the All Network Backbone Routers Multicast Address.
     *
     * @returns A reference to the All Network Backbone Routers Multicast Address.
     *
     */
    const Ip6::Address &GetAllNetworkBackboneRoutersAddress(void) const
    {
        return mAllNetworkBackboneRouters.GetAddress();
    }
```

If `mAllNetworkBackboneRouters` is only properly set up when BBR is enabled, should this method returns `OT_ERROR_INVALID_STATE` when BBR is not enabled since  `mAllNetworkBackboneRouters` is incorrect in such situation?

Reply:
We should have already had similar discussion in #4795
https://github.com/openthread/openthread/pull/4795#discussion_r407312005